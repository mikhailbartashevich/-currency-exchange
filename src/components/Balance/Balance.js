import React from 'react';
import './Balance.css';
import { pure } from 'recompose';
import PropTypes from 'prop-types';

export const BalanceRaw = props => (
  <span className="Balance">
    <span>
      {props.amount} {props.currency ? props.currency.symbol : ''}
    </span>
  </span>
);

BalanceRaw.propTypes = {
  amount: PropTypes.number,
  currency: PropTypes.shape({
    currency: PropTypes.string,
    symbol: PropTypes.string,
  }),
};

export const Balance = pure(BalanceRaw);
