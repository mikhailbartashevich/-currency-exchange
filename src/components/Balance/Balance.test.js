import React from 'react';
import { shallow } from 'enzyme';
import { Balance } from './Balance';

describe('Balance', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Balance debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const component = shallow(<Balance />);
    expect(component).toMatchSnapshot();
  });

  it('should render balance with given props', () => {
    const inputCurrency = { currency: 'USD', symbol: '$' };
    const component = shallow(<Balance amount={10} currency={inputCurrency} />);
    expect(component).toMatchSnapshot();
  });
});
