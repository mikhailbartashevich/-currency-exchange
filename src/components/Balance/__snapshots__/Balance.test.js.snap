// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance should render balance with given props 1`] = `
<BalanceRaw
  amount="10"
  currency={
    Object {
      "currency": "USD",
      "symbol": "$",
    }
  }
/>
`;

exports[`Balance should render correctly in "debug" mode 1`] = `
<BalanceRaw
  debug={true}
/>
`;

exports[`Balance should render correctly with no props 1`] = `<BalanceRaw />`;
